simplejson/_speedups.c

```diff
@@ -277,6 +279,37 @@ moduleinit(void);
 
 #define MIN_EXPANSION 6
 
+static PyObject* JsonStrType;
+#if PY_MAJOR_VERSION < 3
+static PyObject* JsonUniType;
+#endif /* PY_MAJOR_VERSION < 3 */
+static int is_json_type(PyObject *pystr)
+{
+    if (JsonStrType == NULL) {
+        PyObject *encoder_module = PyImport_ImportModule("simplejson.encoder");
+        JsonStrType = PyObject_GetAttrString(encoder_module, "_JsonStr");
+        Py_DECREF(encoder_module);
+        if (JsonStrType == NULL)
+            return 0;
+    }
+    if (PyObject_isInstance(pystr, JsonStrType))
```

Comment:
Interesting, you're correct it is `PyObject_IsInstance`, however I had tested before by adding `return 0;` to the top of the function and it gave me the old result, then removing it gave me the new result. I just tried it again with the same results. Either way, I will fix this.